### PR TITLE
Staging: create s3 bucket for static website with cloudfront distribution

### PR DESCRIPTION
**Pull request by n1tranquilla**
Not included in this MR:
* IAM configuration for CI/CD user
* Certificates for CloudFront via ACM
* Production setup.

These will come later.

### CODE REVIEW

File: tf/environments/staging/minasearch/main.tf [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+}
+
+provider "aws" {
+  region = local.region
+}
+
+module "remote_state" {
+  source      = "../../../modules/minasearch/"
+  bucket_name = local.bucket_name
+}
```
Comment (trevorbernard) [inline tf/environments/staging/minasearch/main.tf:25]: This should be encrypted and preferable in s3. I have an example in the repo

Comment (n1tranquilla) [reply tf/environments/staging/minasearch/main.tf]: Addressed.

File: tf/environments/staging/mina-static-website/main.tf [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+      version = ">= 4.0.0"
+    }
+  }
+  backend "s3" {
+    bucket         = "granola-tfstate-prod"
+    key            = "mina-static-website/terraform.tfstate"
+    region         = "ca-central-1"
+    encrypt        = true
+    kms_key_id     = "alias/state-key-prod"
+    dynamodb_table = "granola-tfstate-lock-prod"
```
Comment (trevorbernard) [inline tf/environments/staging/mina-static-website/main.tf:22]: Same as the first comment. This likely should be staging vs prod

**Review by trevorbernard — CHANGES_REQUESTED** [review]

File: tf/environments/staging/mina-static-website/main.tf [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
+    }
+  }
+  backend "s3" {
+    bucket         = "granola-tfstate-prod"
+    key            = "mina-static-website/terraform.tfstate"
```
[the remark this answers is not in the document]
Comment (n1tranquilla) [reply tf/environments/staging/mina-static-website/main.tf:18]: I don't think so. I've changed the folder to `mina-static-website` as an instance of `s3-static-website`

File: tf/environments/staging/mina-static-website/main.tf [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+  required_version = ">= 1.0.0"
+
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.0.0"
+    }
+  }
+  backend "s3" {
+    bucket         = "granola-tfstate-prod"
```
[the remark this answers is not in the document]
Comment (n1tranquilla) [reply tf/environments/staging/mina-static-website/main.tf:17]: Do we have a staging bucket yet?

File: tf/environments/staging/mina-static-website/main.tf [hunk 5]
```diff
@@ -0,0 +1,37 @@
+locals {
+  user_name   = "terraformuser"
```
Comment (trevorbernard) [inline tf/environments/staging/mina-static-website/main.tf:2]: remove this line

File: tf/environments/staging/mina-static-website/main.tf (hunk 3 above)
[the remark this answers is not in the document]
Comment (n1tranquilla) [reply tf/environments/staging/mina-static-website/main.tf:18]: Maybe this needs to be `staging-mina-static-website`.

[the remark this answers is not in the document]
Comment (trevorbernard) [reply tf/environments/staging/mina-static-website/main.tf]: I don't think `mina-static-website` is descriptive or unique enough.

**Review by trevorbernard — APPROVED** [review]

**Pull request merged**